mlx_graphs/transforms/base_transform.py

```diff
@@ -0,0 +1,21 @@
+import copy
+from abc import ABC
+from typing import Any
+
+
+class BaseTransform(ABC):
+    r"""An abstract class for creating transformations
+
+    Transforms are a way to modify and customize
+    ``GraphData`` by implicitly passing them as arguments
+    in ``Dataset``
+    """
+
+    def __call__(self, data: Any) -> Any:
+        return self.process(copy.copy(data))
```

Comment:
why is the copy needed here?

Reply:
So that we don't accidently change the values inplace

Reply:
ok, makes sense for when it's passed to `transform` instead of `pre_transform` 👍